src/editTask.js

```diff
@@ -0,0 +1,84 @@
+const todoList = document.querySelector('.todo-list');
+
+const toggleComplete = (inputElement) => {
+  if (inputElement.checked === false) {
+    inputElement.parentElement.parentElement.classList.remove('strike-thro');
+    inputElement.parentElement.parentElement.parentElement.classList.remove('active');
+  } else {
+    inputElement.parentElement.parentElement.classList.add('strike-thro');
+    inputElement.parentElement.parentElement.parentElement.classList.add('active');
+  }
+};
+
+const showEditInputs = (liElement) => {
+  const editInput = document.querySelector('.edit-input');
+  if (editInput) {
+    editInput.remove();
+  }
+  const input = document.createElement('input');
+  input.type = 'text';
+  input.name = 'editInput';
+  input.value = liElement.textContent;
+  input.classList.add('edit-input');
+  liElement.appendChild(input);
+  input.focus();
+};
+// works with toggleComplete
+function editTask() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('change', (e) => {
+        e.stopPropagation();
+        if (e.target.tagName === 'INPUT' && e.target.type === 'checkbox') {
+          toggleComplete(e.target);
+        }
+      });
+    });
+  });
+}
+
+function editText() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('click', (e) => {
+        e.stopPropagation();
+        if (e.target.tagName === 'LI') {
+          showEditInputs(e.target);
+        }
+      });
+    });
+  });
+}
+
+function updateText() {
+  const editInput = document.querySelector('.edit-input');
+  if (!editInput) {
+    return;
+  }
+  const newText = editInput.value;
+  if (newText !== '') {
+    const liInput = editInput.parentElement;
+    liInput.textContent = newText;
+  }
+  editInput.remove();
+}
+
+function keyPress() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('keypress', (e) => {
+        //  e.stopPropagation();
+        if (e.target.tagName === 'INPUT' && e.target.type === 'text' && e.key === 'Enter') {
+          updateText();
+        }
+      });
+    });
+  });
+}
```

Comment:
- [ ] Kindly your edit method works fine, but you should save the updated data to the local storage so when you reload the page you find the updated data.

![image](https://user-images.githubusercontent.com/35535130/160959491-99c35e64-67b2-4dc4-8d85-071bd41a41e3.png)